src/services/todo-lists-service.js

```diff
@@ -7,65 +7,130 @@ const Utils = require('../utils');
 
 class TodoListsService {
 
+    // ---------------------- REPOSITORY SERVICES ----------------------//
+
     checkAndSetAvailability(id) {
         return listRepository.checkAndSetAvailability(id);
     }
 
-    createList(elements) {
-        return this.performAction(null, nodesToCommit => {
-            // TODO HERE WE NEED TO MAKE THE CALL TO THE OTHER NODES AND RETURN THE LIST WITH THE APPLIED CHANGES
-        })
+    createList(list) {
+        return listRepository.createList(list);
+    }
+
+    updateAndUnlockList(id, list) {
+        let updatedList = listRepository.updateList(id, list);
+        listRepository.checkAndSetAvailability(id, true);
+        return updatedList;
+    }
+
+    addItem(id, item) {
+        return listRepository.addItem(id, item);
+    }
+
+    updateItem(id, index, item) {
+        return listRepository.updateItem(id, index, item);
+    }
+
+    updateItemReadyStatus(id, index, ready) {
+        return listRepository.updateItemReadyStatus(id, index, ready);
+    }
+
+    updateItemPosition(id, index, newIndex) {
+        return listRepository.updateItemPosition(id, index, newIndex);
+    }
+
+    deleteItem(id, index) {
+        return listRepository.deleteItem(id, index);
+    }
+
+    // ---------------------- COMMIT SERVICES ----------------------//
+
+    performCreateList(newList) {
+        // For the list creation we make a commit without quorum.
+
+        let createdList = this.createList(newList);
+        let nodes = nodesService.getAllButSelf();
+
+        // We give a null listId because we are creating it.
+        let list = this.commitToNodes(nodes, null, createdList);
+
+        return this.ok(list);
     }
 
-    addElement(listId, element) {
+    performAddItem(listId, item) {
         return this.performAction(listId, nodesToCommit => {
```

Comment:
```suggestion
        return this.performAction(listId, _ => {
```
Habría que eliminar esto en todos los métodos porque al final no se usa

Reply:
que colgada :new_moon_with_face: